### ISSUE TITLE
Input dimensions mis-match?

### ISSUE THREAD

**Issue by emrahgunduz** [deepfakes/faceswap-playground#192]
## Expected behavior

I just wanted to give a try before diving deep, and downloaded Cage/Trump data. I simply cut-paste a few images to see the speed of the system on my GPU.

## Actual behavior

Command line used is:
`python faceswap.py extract -i D:\Projects\faceswap\data\photos\test -o D:\Projects\faceswap\data\data\test -v`

With the default Cage set I'm receiving a "Input dimension mis-match" error, and on the second image error becomes "CUDA_ERROR_OUT_OF_MEMORY". Was wondering how could I solve this problem?

First image:
```
Failed to extract from image: D:\Projects\faceswap\data\photos\test\1.jpeg. Reason: GpuElemwise. Input dimension mis-match. Input 1 (indices start at 0) has shape[2] == 1, but the output's size on that axis is 128.
Apply node that caused the error: GpuElemwise{mul,no_inplace}(GpuElemwise{sub,no_inplace}.0, GpuElemwise{true_div,no_inplace}.0)
Toposort index: 125
Inputs types: [GpuArrayType<None>(float32, 4D), GpuArrayType<None>(float32, 4D)]
Inputs shapes: [(1, 64, 128, 128), (1, 64, 1, 1)]
Inputs strides: [(4194304, 65536, 512, 4), (256, 4, 4, 4)]
Inputs values: ['not shown', 'not shown']
Inputs type_num: [11, 11]
Outputs clients: [[GpuElemwise{mul,no_inplace}(GpuElemwise{mul,no_inplace}.0, GpuReshape{4}.0)]]
```

Second image:
```
Failed to extract from image: D:\Projects\faceswap\data\photos\test\2.jpeg. Reason: b'cuMemAlloc: CUDA_ERROR_OUT_OF_MEMORY: out of memory'
Apply node that caused the error: GpuElemwise{mul,no_inplace}(GPUA_mrg_uniform{GpuArrayType<None>(float32, 4D),no_inplace}.1, GpuArrayConstant{[[[[0.11785113]]]]})
Toposort index: 1
Inputs types: [GpuArrayType<None>(float32, 4D), GpuArrayType<None>(float32, (True, True, True, True))]
Inputs shapes: [(3, 3, 128, 64), (1, 1, 1, 1)]
Inputs strides: [(98304, 32768, 256, 4), (4, 4, 4, 4)]
Inputs values: ['not shown', gpuarray.array([[[[0.11785113]]]], dtype=float32)]
Inputs type_num: [11, 11]
Outputs clients: [[GpuElemwise{add,no_inplace}(GpuElemwise{mul,no_inplace}.0, GpuArrayConstant{[[[[-0.05892557]]]]})]]
```

I tried using smaller images (128, 64, etc), bigger images (512, 1024) but the errors are always the same.

**OS:** Windows 10
**python --version** Python 3.6.6 :: Anaconda, Inc.
**Faceswap version:** 9696b5606fd0963814fc0c3644565aa60face69d
**Faceswap method:** GPU

| Stats        |                     |
| ------------ | ------------------- |
| driver       | 398.11              |
| device count | 1                   |
| devices      | ['GeForce GTX 970'] |
| vram         | [4096.0]            |
| vram free    | 3279.23046875       |

**pip freeze**
```
absl-py==0.2.2
alabaster==0.7.11
asn1crypto==0.24.0
astor==0.6.2
Babel==2.6.0
bleach==1.5.0
certifi==2018.4.16
cffi==1.11.5
chardet==3.0.4
click==6.7
cloudpickle==0.5.3
colorama==0.3.9
cryptography==2.2.2
cycler==0.10.0
Cython==0.28.3
dask==0.18.1
decorator==4.3.0
dlib==19.10.99
docutils==0.14
face-recognition==1.2.2
face-recognition-models==0.3.0
ffmpy==0.2.2
gast==0.2.0
grpcio==1.12.1
h5py==2.8.0
html5lib==0.9999999
idna==2.7
imageio==2.3.0
imagesize==1.0.0
Jinja2==2.10
Keras==2.2.0
Keras-Applications==1.0.2
Keras-Preprocessing==1.0.1
kiwisolver==1.0.1
mako==1.0.7
Markdown==2.6.11
MarkupSafe==1.0
matplotlib==2.2.2
mkl-fft==1.0.0
mkl-random==1.0.1
networkx==2.1
nose==1.3.7
numpy==1.14.5
nvidia-ml-py3==7.352.0
olefile==0.45.1
opencv-python==3.4.1.15
packaging==17.1
pathlib==1.0.1
Pillow==5.1.0
protobuf==3.5.2
pycparser==2.18
Pygments==2.2.0
pygpu==0.7.6
pyOpenSSL==18.0.0
pyparsing==2.2.0
PySocks==1.6.8
python-dateutil==2.7.3
pytz==2018.5
PyWavelets==0.5.2
PyYAML==3.12
requests==2.19.1
scandir==1.5
scikit-image==0.14.0
scipy==1.1.0
six==1.11.0
snowballstemmer==1.2.1
Sphinx==1.7.5
sphinxcontrib-websupport==1.1.0
tensorboard==1.9.0
tensorflow==1.8.0
tensorflow-gpu==1.9.0
termcolor==1.1.0
Theano==1.0.2+2.gc449c8699
toolz==0.9.0
tornado==5.0.2
tqdm==4.23.4
urllib3==1.23
Werkzeug==0.14.1
win-inet-pton==1.0.1
wincertstore==0.2
```

**Comment by gessyoo** [comment]
My setup is very similar to yours, Windows 10, Ananconda python 3.65, except a GTX 1070 and I'm using Tensorflow 1.8.  I think the code was originally built with TF 1.5 and CUDA 9.0 as requirements, maybe try pip install -r requirements.txt, in case of a bug in some of the newer versions of the python components.   Also, aren't the Cage and Trump face sets already extracted and aligned 256x256 images?  You don't need to extract those faces again.  Try training with those images to test your setup.  Then try the extract process on a different set of video frames with faces in them.  I've mostly used 1280x720 images, and I haven't had the error you're reporting.  You've installed the CUDA 9.0 toolkit and the CUDA dnn files, I presume?

**Comment by emrahgunduz** [comment]
Sorry for the late reply.

Yes I've CUDA 9.0 and cudnn files installed, usin TF 1.8.
I used requirements to setup, but dlib was problematic on Windows. I installed that from a wheel (19.10.99) first. It looks like the setup is correct.

Will try using with other images with faces instead of the default data. Just wanted to test if it would crop with the same size. Will also try to teach with Cage/Trump data first, and let you know.
